source/unified-test-format/schema-1.2.json

```diff
@@ -0,0 +1,293 @@
+      "minProperties": 1,
+      "properties": {
+        "client": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["id"],
+          "properties": {
+            "id": { "type": "string" },
+            "uriOptions": { "type": "object" },
+            "useMultipleMongoses": { "type": "boolean" },
+            "observeEvents": {
+              "type": "array",
+              "minItems": 1,
+              "items": {
+                "type": "string",
+                "enum": ["commandStartedEvent", "commandSucceededEvent", "commandFailedEvent"]
+              }
+            },
+            "ignoreCommandMonitoringEvents": { "$ref": "#/definitions/ignoreCommandMonitoringEvents" }
+          }
+        },
+        "database": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["id", "client", "databaseName"],
+          "properties": {
+            "id": { "type": "string" },
+            "client": { "type": "string" },
+            "databaseName": { "type": "string" },
+            "databaseOptions": { "$ref": "#/definitions/collectionOrDatabaseOptions" }
+          }
+        },
+        "collection": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["id", "database", "collectionName"],
+          "properties": {
+            "id": { "type": "string" },
+            "database": { "type": "string" },
+            "collectionName": { "type": "string" },
+            "collectionOptions": { "$ref": "#/definitions/collectionOrDatabaseOptions" }
+          }
+        },
+        "session": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["id", "client"],
+          "properties": {
+            "id": { "type": "string" },
+            "client": { "type": "string" },
+            "sessionOptions": { "type": "object" }
+          }
+        },
+        "bucket": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["id", "database"],
+          "properties": {
+            "id": { "type": "string" },
+            "database": { "type": "string" },
+            "bucketOptions": { "type": "object" }
+          }
+        },
+        "stream": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["id", "hexBytes"],
+          "properties": {
+            "id": { "type": "string" },
+            "hexBytes": {
+              "type": "string",
+              "pattern": "^([0-9A-Fa-f]{2})*$"
+            }
+          }
+        }
+      }
+    },
+
+    "collectionData": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["collectionName", "databaseName", "documents"],
+      "properties": {
+        "collectionName": { "type": "string" },
+        "databaseName": { "type": "string" },
+        "collectionOptions": { "type": "object" },
```

Comment:
The `collectionData` definition is used for both `initialData` as well as `outcome` in tests. I would suggest splitting this up into `initialData` and `outcomeData` definitions, as specifying `collectionOptions` in an `outcome` field should not be possible.

Reply:
Done. I split them into `initialCollectionData` and `outcomeCollectionData` types.

Reply:
When test runners create a collection on the server, they use a test file's `initialData` directive. This directive accepts collection/database names, not entity references, so it doesn't actually know what options were used to configure the entity. Note that it can't be changed to accept entity references because the changes in this PR allow test files to create new entities at the test-level, so the collection being referenced might not even exist as an entity yet.

Also, we discussed some workarounds to try to reuse the existing `collectionOptions` stuff, but ultimately decided that it's purpose is to declare options to configure a client-side `MongoCollection` object, not a server-side collection.